applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -587,19 +586,8 @@ static void stream_recv_cb(struct bt_bap_stream *stream, const struct bt_iso_rec
 		return;
 	}
 
-	uint32_t octets_per_frame = bt_codec_cfg_get_octets_per_frame(stream->codec);
-
-	if (buf->len != octets_per_frame && bad_frame != true) {
-		data_size_mismatch_cnt++;
-		if ((data_size_mismatch_cnt % 500) == 1) {
-			LOG_DBG("Data size mismatch, received: %d, codec: %d, total: %d", buf->len,
-				octets_per_frame, data_size_mismatch_cnt);
-		}
-
-		return;
-	}
-
-	receive_cb(buf->data, buf->len, bad_frame, info->ts, channel_index);
+	receive_cb(buf->data, buf->len, bad_frame, info->ts, channel_index,
+		   bt_codec_cfg_get_octets_per_frame(stream->codec));
```

Comment:
Since we know bt_codec_cfg_get_octets_per_frame() takes much time, we should at least have a task for changing it

Reply:
Both for cis_gateway and cis_headset

Reply:
I believe you are already working on this @rick1082 ?

Reply:
Yes, I'm working on a new module called by codec_helper, which help us parsing the codec info, and also there will be a structure for storing the current codec setting, so we don't need to parse it every time.